FirebaseInstallations/Source/Tests/Unit/FIRInstallationsTests.m

```diff
@@ -289,6 +278,21 @@ - (void)testInitWhenAppOptionsMissingThenThrows {
   XCTAssertThrows([self createInstallationsWithAppOptions:nil appName:@"missingOptions"]);
 }
 
+- (void)testInitWithAPIKeyIsNotMatchingExpectedFormat {
+  FIROptions *options = [self.appOptions copy];
+  options.APIKey = @"ATooShortKey00000000000000000000000000";
+  XCTAssertThrows([self createInstallationsWithAppOptions:options appName:@"shortAPIKey"]);
+
+  options.APIKey = @"ATooLongKey00000000000000000000000000000";
+  XCTAssertThrows([self createInstallationsWithAppOptions:options appName:@"longAPIKey"]);
+
+  options.APIKey = @"StartsNotFromA0000000000000000000000000";
+  XCTAssertThrows([self createInstallationsWithAppOptions:options appName:@"wrongFirstCharacter"]);
+
+  options.APIKey = @"A ContainsNotAlphaNumeric-+=00000000000";
+  XCTAssertThrows([self createInstallationsWithAppOptions:options appName:@"invalidCharacters"]);
```

Comment:
Alternative suggestions. \
Please feel free to ignore.

* "`AIzaSy-ApiKeyWithValidFormat_0123456789`"
* "`AIzaSy-ApiKeyInvalidLength_0123456789`"
* "`AIzaSy-ApiKeyWithLengthTooLong_0123456789`"
* "`BBAIzaSy-ApiKeyInvalidFormat_0123456789`"
* "`AIzaSy-+-ApiKeyInvalidFormat_0123456789`"